unity/Assets/Scripts/Util/Algorithms/Polygon/Martinez.cs

```diff
@@ -0,0 +1,984 @@
+
+                // Delete line segment associated to "ev" from status and check for intersection between the neighbors of "ev" in status
+                status.Delete(ev.StatusItem);
+
+                if (nextFound && prevFound)
+                {
+                    PossibleIntersection(prev.SweepEvent, next.SweepEvent, events);
+                }
+            }
+        }
+
+        private void ComputeFields(SweepEvent ev, StatusItem prev, bool prevFound)
+        {
+            // Compute InOut and OtherInOut fields
+            if (!prevFound)
+            {
+                ev.InOut = false;
+                ev.OtherInOut = true;
+            }
+            else if (ev.PolygonType == prev.SweepEvent.PolygonType
+            ) // Previous line segment in status belongs to the same polygon that "ev" belongs to
+            {
+                ev.InOut = !prev.SweepEvent.InOut;
+                ev.OtherInOut = prev.SweepEvent.OtherInOut;
+            }
+            else // Previous line segment in status belongs to a different polygon that "ev" belongs to
+            {
+                ev.InOut = !prev.SweepEvent.OtherInOut;
+                ev.OtherInOut = prev.SweepEvent.Vertical ? !prev.SweepEvent.InOut : prev.SweepEvent.InOut;
+            }
+
+            // Compute PreviousInResult field
+            if (prevFound)
+            {
+                ev.PreviousInResult = (!InResult(prev.SweepEvent) || prev.SweepEvent.Vertical)
+                    ? prev.SweepEvent.PreviousInResult
+                    : prev.SweepEvent;
+            }
+
+            // Check if the line segment belongs to the boolean operation
+            ev.InResult = InResult(ev);
+        }
+
+        private bool InResult(SweepEvent ev)
+        {
+            switch (ev.EdgeType)
+            {
+                case EdgeType.Normal:
+                    switch (Operation)
+                    {
+                        case OperationType.Intersection:
+                            return !ev.OtherInOut;
+                        case OperationType.Union:
+                            return ev.OtherInOut;
+                        case OperationType.Difference:
+                            return (ev.PolygonType == PolygonType.Subject && ev.OtherInOut) ||
+                                   (ev.PolygonType == PolygonType.Clipping && !ev.OtherInOut);
+                        case OperationType.Xor:
+                            return true;
+                    }
+
+                    break;
+                case EdgeType.SameTransition:
+                    return Operation == OperationType.Intersection || Operation == OperationType.Union;
+                case EdgeType.DifferentTransition:
+                    return Operation == OperationType.Difference;
+                case EdgeType.NonContributing:
+                    return false;
+            }
+
+            // Just to make it compile
+            return false;
```

Comment:
Please remove in the final version

Reply:
I can't because then it doesn't compile. Apparently C# doesn't recognize that the switch statement is exhaustive.